packages/messaging/lib/index.js

```diff
@@ -311,6 +330,22 @@ class FirebaseMessagingModule extends FirebaseModule {
     }
   }
 
+  setOpenSettingsForNotificationsHandler(handler) {
+    if (!isIOS) {
+      throw new Error(
+        'firebase.messaging().setOpenSettingsForNotificationsHandler() is only supported on iOS devices.',
+      );
+    }
```

Comment:
This is an interesting choice, as it forces platform-specific conditionals on API consumers, for no real benefit I think. Do we follow this pattern elsewhere so this is really necessary to keep the APIs working the same as the rest of the library, or could we just silently return if it's not iOS? I'd lean towards just silently dropping it on android if possible

Reply:
Yes you're right, let me fix that :)))